src/coreclr/jit/gentree.cpp

```diff
@@ -19978,17 +19978,21 @@ bool GenTree::SupportsSettingZeroFlag()
     }
 #endif
 #elif defined(TARGET_ARM64)
-    if (OperIs(GT_AND, GT_AND_NOT, GT_NEG))
+    if (OperIs(GT_AND, GT_AND_NOT))
     {
         return true;
     }
 
     // We do not support setting zero flag for madd/msub.
+    if (OperIs(GT_NEG) && (!gtGetOp1()->OperIs(GT_MUL) || !gtGetOp1()->isContained()))
```

Comment:
I presume just `if (OperIs(GT_NEG) && !gtGetOp1()->isContained())` would be enough since NEG can't contain anything. Also I presume we should've had a separate node type for MUL+ADD/SUB